module-apps/application-messages/windows/SMSThreadViewWindow.cpp

```diff
@@ -62,29 +63,34 @@ namespace gui
 
     void SMSThreadViewWindow::onBeforeShow(ShowMode mode, SwitchData *data)
     {
-        {
-            auto pdata = dynamic_cast<SMSThreadData *>(data);
-            if (pdata) {
-                LOG_INFO("Thread data received: %" PRIu32, pdata->thread->ID);
-                requestContact(pdata->thread->numberID);
-
-                // Mark thread as Read
-                if (pdata->thread->isUnread()) {
-                    auto app = dynamic_cast<app::ApplicationMessages *>(application);
-                    assert(app != nullptr);
-                    if (application->getCurrentWindow() == this) {
-                        app->markSmsThreadAsRead(pdata->thread->ID);
-                    }
+        if (auto pdata = dynamic_cast<SMSThreadData *>(data); pdata) {
+            LOG_INFO("Thread data received: %" PRIu32, pdata->thread->ID);
+            requestContact(pdata->thread->numberID);
+
+            // Mark thread as Read
+            if (pdata->thread->isUnread()) {
+                auto app = dynamic_cast<app::ApplicationMessages *>(application);
+                assert(app != nullptr);
+                if (application->getCurrentWindow() == this) {
+                    app->markSmsThreadAsRead(pdata->thread->ID);
                 }
-                smsModel->numberID    = pdata->thread->numberID;
-                smsModel->smsThreadID = pdata->thread->ID;
-                smsList->rebuildList();
-            }
-            else if (smsModel->numberID != DB_ID_NONE) {
-                requestContact(smsModel->numberID);
             }
+            smsModel->numberID    = pdata->thread->numberID;
+            smsModel->smsThreadID = pdata->thread->ID;
+            smsList->rebuildList();
+
+            ssize_t countToClear = -pdata->thread->unreadMsgCount;
```

Comment:
if it is ssize_t please check if not negative before clearing.

Reply:
`countToClear` is expected to be a negative value of `pdata->thread->unreadMsgCount`

Reply:
This is reworked now - added special query to handle decrementing of notification count.